### DISCUSSION TITLE
How to put a button in the gridcolumns?

### DISCUSSION THREAD

**Discussion by LeandroSiani** [vikramlearning/blazorbootstrap#946] [Q&A] [answered]
I would like to know how to create a grid with Edit/remove events in each row.

**Comment by diegosyswork** [comment, marked as answer]
Hi, you can use <ContentChildren> on a new column and put whatever comes to mind there, in my case I created a component that has the buttons: show, edit and delete             
            
            <GridColumn TItem="ViewCustomersdb" PropertyName="Periodicity" Filterable="false" Sortable="false">
                <HeaderContent>
                </HeaderContent>
                <ChildContent>
                    <ShowEditDeleteToolbar 
                        HrefShow=@(_hrefShow + "/" + context.CustomerDbID)
                        HrefEdit=@(_hrefEdit + "/" + context.CustomerDbID)
                        HrefDelete=@(_hrefDelete + "/" + context.CustomerDbID) />
                </ChildContent>
            </GridColumn>


create a new class named ShowEditDeleteToolbar.razor: with this code:

 <div class="d-flex" >

    @if (!string.IsNullOrEmpty(HrefShow))
    {
        <a href="@(HrefShow)" class="text-decoration-none actionToolbarEditDelete">
            <Tooltip Title="Ver" role="button">
                <Icon Size="IconSize.x5" Name="IconName.Eye"></Icon>
            </Tooltip>
        </a>
    }
    @if (!string.IsNullOrEmpty(HrefEdit))
    {
        <a href="@(HrefEdit)" class="text-decoration-none actionToolbarEditDelete">
            <Tooltip Title="Editar" role="button">
                <Icon Size="IconSize.x5" Name="IconName.Pencil"></Icon>
            </Tooltip>
        </a>
    }
    @if (!string.IsNullOrEmpty(HrefDelete))
    {
        <a href="@(HrefDelete)" class="text-decoration-none actionToolbarEditDelete">
            <Tooltip Title="Eliminar" role="button">
                <Icon Size="IconSize.x5" Name="IconName.Trash"></Icon>
            </Tooltip>
        </a>
    }
</div>


@code
{
    [Parameter]
    public string? HrefShow { get; set; } = null;
    
    [Parameter]
    public string? HrefEdit { get; set; } = null;

    [Parameter]
    public string? HrefDelete { get; set; } = null;

    [Parameter]
    public object? Tag{ get; set; } = null;
}


I hope I have helped you.

**Reply by LeandroSiani** [reply]
Hi, tks for help!
It's worked!
